### PR TITLE
Update Group Samples Table

### LINKED ISSUE #266: Update: Group Samples table
**Problem**

Similar to #265 the group samples table only displays the sample name and description. 

### Solution

**Update columns to include the following:**

1.  Name, _include the group hierarchy, but try using only the first letter `p/y/pyogenes-001` and then on hover have a popover with the full path._
2.  Created Date
3.  Last Updated Date
4.  Other metadata associated with the sample.  This should not be loaded on the first load but through a dropdown or other selection mechanism.

**Search functionality**

Start simple by only searching the name.  Maybe expand into a larger search later.

But not in a card

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Made the following changes to the group samples table:
- Updated `Project` column to include the group hierarchy using only the first letter with a popover that displays the full path on hover
- Added `Created` column
- Added `Last Updated` column
- Added sorting by name or date
- Added search/filter by name

*Metadata not included.*

Fixes #266.

## Screenshots or screen recordings
image

## How to set up and validate locally
1. Navigate to a group.
2. View samples.
3. Verify filtering and sorting works.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by joshsadam — CHANGES_REQUESTED** [review]
This works exactly as it is supposed to.  I just have one request and that this to add a test for the abbreviated path.  Great work.

**Review by ChrisHuynh333 — DISMISSED** [review]
This works as expected and looks good to me. Only thing is with the abbreviated path, would it be more beneficial to have maybe 3 characters so it's a little more recognizable at a glance for the user? Unless we expect lots of namespace nesting by users or the table to be further crowded (metadata?), then it's understandable to have it as a single char. I'll let someone else have final word on this, just a thought!

File: app/models/concerns/routable.rb [hunk 1]
```diff
@@ -49,6 +49,19 @@ def full_path
     end
   end
 
+  def abbreviated_path
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/models/concerns/routable.rb:52]: I agree, specifically this should be in a model test.

[the remark this answers is not in the document]
Comment (ksierks) [reply app/models/concerns/routable.rb:52]: Is this sufficient? Please see 9bd1e197e5d30849c2580cba108f3ba26f0ae89d.

**Comment by ksierks** [comment]
> This works as expected and looks good to me. Only thing is with the abbreviated path, would it be more beneficial to have maybe 3 characters so it's a little more recognizable at a glance for the user? Unless we expect lots of namespace nesting by users or the table to be further crowded (metadata?), then it's understandable to have it as a single char. I'll let someone else have final word on this, just a thought!

It makes no difference to me. I only used the first character as that is what was described in the issue. Please let me know if I should change it to 3 characters.

**Comment by ericenns** [comment]
> > This works as expected and looks good to me. Only thing is with the abbreviated path, would it be more beneficial to have maybe 3 characters so it's a little more recognizable at a glance for the user? Unless we expect lots of namespace nesting by users or the table to be further crowded (metadata?), then it's understandable to have it as a single char. I'll let someone else have final word on this, just a thought!
> 
> It makes no difference to me. I only used the first character as that is what was described in the issue. Please let me know if I should change it to 3 characters.

We are going to stick with 1 character to represent each higher level, as the full hierarchy is viewable on mouse over.

File: app/models/concerns/routable.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ksierks) [reply app/models/concerns/routable.rb:52]: And 6c9ebdb425355a3d4152913c5e4995a7c1a3cecd.

[the remark this answers is not in the document]
Comment (joshsadam) [reply app/models/concerns/routable.rb:52]: Thanks

**Review by joshsadam — APPROVED** [review]
This looks good to me.

**Review by ericenns — APPROVED** [review]
This looks good to me!

**Pull request merged**